src/nvim/macros.h

```diff
@@ -59,12 +59,12 @@
 /* Use our own isdigit() replacement, because on MS-Windows isdigit() returns
  * non-zero for superscript 1.  Also avoids that isdigit() crashes for numbers
  * below 0 and above 255.  */
-#define VIM_ISDIGIT(c) ((unsigned)(c) - '0' < 10)
+#define VIM_ISDIGIT(c) ((unsigned)(c) >= '0' && (unsigned)(c) < '0' + 10)
 
 /* Like isalpha() but reject non-ASCII characters.  Can't be used with a
  * special key (negative value). */
-# define ASCII_ISLOWER(c) ((unsigned)(c) - 'a' < 26)
-# define ASCII_ISUPPER(c) ((unsigned)(c) - 'A' < 26)
+# define ASCII_ISLOWER(c) ((unsigned)(c) >= 'a' && (unsigned)(c) < 'a' + 26)
+# define ASCII_ISUPPER(c) ((unsigned)(c) >= 'A' && (unsigned)(c) < 'A' + 26)
```

Comment:
I'm not sure if it's in the scope of the PR, but I think <= 'z' (and <= '9' above) would look clearer.